Source/WebGPU/WebGPU/Pipeline.mm

```diff
@@ -191,16 +191,16 @@
             uint64_t bufferSize = 0;
             if (auto* bufferBinding = get_if<WGPUBufferBindingLayout>(&it->value.bindingLayout))
                 bufferSize = bufferBinding->minBindingSize;
-            if (!bufferSize && mininumBufferSizes) {
+            if (mininumBufferSizes) {
                 if (auto bufferSizeIt = mininumBufferSizes->find(it->value.binding); bufferSizeIt != mininumBufferSizes->end())
-                    bufferSize = bufferSizeIt->value;
+                    bufferSize = std::max(bufferSize, bufferSizeIt->value);
```

Comment:
I'm not sure if I'm interpreting the spec correctly, but shouldn't it be an error if `bufferSizeIt->value > bufferSize`? From the [spec](https://www.w3.org/TR/webgpu/#dom-gpubufferbindinglayout-minbindingsize):

```If this is not 0, pipeline creation additionally [validates](https://www.w3.org/TR/webgpu/#abstract-opdef-validating-shader-binding) that this value ≥ the [minimum buffer binding size](https://www.w3.org/TR/webgpu/#minimum-buffer-binding-size) of the variable.``

Reply:
oh nice observation, I should add that condition too. Will update the PR